test/terraform_aws_endpoint_example_test.go

```diff
@@ -0,0 +1,75 @@
+package test
+
+import (
+	"os"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+// An example of how to test the Terraform module in examples/terraform-aws-endpoint-example using Terratest.
+func TestTerraformAwsEndpointExample(t *testing.T) {
+	t.Parallel()
+
+	// Set a custom endpoint for AWS, and set the keys to dummy keys to
+	// pass that check
+	os.Setenv("TERRATEST_CUSTOM_AWS_ENDPOINT", "http://localhost:5000")
```

Comment:
I think should be better to use ``t.Setenv()` since it has automatic cleanup of env variables

https://pkg.go.dev/testing#B.Setenv

Reply:
I get this error:

```
	panic: testing: t.Setenv called after t.Parallel; cannot set environment variables in parallel tests 
```

If I make it a non parallel test it passes.

Reply:
Oh, in the end I did change it to `t.Setenv`. I just wanted to let you know that had to change it to a non parallel test for that to work.